### PR TITLE
doc/website/blog: entoas blog post

### PR DESCRIPTION
**Pull request by masseelch**
cc @rotemtam @a8m

### CODE REVIEW

**Review by rotemtam — COMMENTED** [review]
Great start. Super exciting! Will review in full tomorrow

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
+---
+title: Announcing entoas 
+author: MasseElch 
+authorURL: "https://github.com/masseelch"
+authorImageURL: "https://avatars.githubusercontent.com/u/12862103?v=4"
+---
+
+OAS (formerly known as Swagger Specification) is a technical specification defining a standard, language-agnostic
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:8]:
```suggestion
The OpenAPI Specification (OAS, formerly known as Swagger Specification) is a technical specification defining a standard, language-agnostic
```

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+interface description for REST APIs. This allows both humans and automated tools to understand the described service
+without the actual source code or additional documentation. Combined with the [Swagger Tooling](https://swagger.io/) you
+can generate both server and client boilerplate code for more than 20 languages, just by passing in the OAS document.
+
+Sometime ago, in a [previous blogpost](https://entgo.io/blog/2021-09-10-openapi-generator.md), we presented to you a new
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:13]:
```suggestion
In a [previous blogpost](https://entgo.io/blog/2021/09/10/openapi-generator), we presented to you a new
```

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+feature of the Ent extension [`elk`](https://github.com/masseelch/elk): a fully
+compliant [OpenAPI Specification (OAS)](https://swagger.io/resources/open-api/) document generator.
+
+Today, we are very happy to announce, that the specification generator is now an official extension to the Ent project
+and has been moved to the [`ent/contrib`](https://github.com/ent/contrib/entoas) repository. In addition, we have
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:18]: contrib link is broken.

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+listened to the feedback of the community and have made some changes to the generator, that you will love.
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:19]:
```suggestion
listened to the feedback of the community and have made some changes to the generator, that we hope
you will like.  
```

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+feature of the Ent extension [`elk`](https://github.com/masseelch/elk): a fully
+compliant [OpenAPI Specification (OAS)](https://swagger.io/resources/open-api/) document generator.
+
+Today, we are very happy to announce, that the specification generator is now an official extension to the Ent project
+and has been moved to the [`ent/contrib`](https://github.com/ent/contrib/entoas) repository. In addition, we have
+listened to the feedback of the community and have made some changes to the generator, that you will love.  
+
+### Getting Started
+
+To use the `entoas` extension you are required to use the `entc` (ent codegen) package as
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:23]:
```suggestion
To use the `entoas` extension use the `entc` (ent codegen) package as
```

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+described [here](https://entgo.io/docs/code-gen#use-entc-as-a-package). First install the extension to your Go module:
+
+```shell
+go get entgo.io/contrib/entoas
+```
+
+Now follow the next two steps to enable it and to configure Ent to work with the `elk` extension:
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:30]: elk?

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+func (Item) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("compartment", Compartment.Type).
+			Ref("contents").
+			Unique(),
+	}
+}
+```
+
+Now run the code generator:
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:151]: write a few sentences explaining what we did in the code block above.  we defined the schema, added these entities, and defined these edges between them yada yada

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+
+```shell
+go generate ./...
+```
+
+In addition to the files Ent normally generates, another file named `ent/openapi.json` has been created. Copy its contents
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:157]: Show the output JSON file ! (can be truncated so its small but should demonstrate the features of the extension without needing to run i t).

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+and paste them into the [Swagger Editor](https://editor.swagger.io/). You should see three groups: **Compartment**, **
+Item** and **Fridge**.
+
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:160]: again show, dont tell (screen shot, please!)

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+### Basic Configuration
+
+The description of our API does not yet reflect what it does, but `entoas` lets you change that! Open up `ent/entc.go`
+and pass in the updated title and description of our Fridge API:
+
+```go title="ent/entc.go"
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:166]: use line highlighting to emphasize whats changing

Comment (rotemtam) [reply doc/website/blog/2021-11-10-announcing-entoas.md:166]: https://docusaurus.io/docs/markdown-features/code-blocks#line-highlighting

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+		log.Fatalf("creating entoas extension: %v", err)
+	}
+	err = entc.Generate("./schema", &gen.Config{}, entc.Extensions(ex))
+	if err != nil {
+		log.Fatalf("running ent codegen: %v", err)
+	}
+}
+```
+
+Rerunning the code generator will create an updated OAS document you can copy-paste into the Swagger Editor.
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:196]: again, show don't tell. show a snippet of the JSON file that shows this change. 
always show cause and effect so people can follow along by example instead

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+to exclude a specific operation by using an `entoas.Annotation`. Policies are used to enabled / disable the generation
+of sub-resource operations as well:
+
+```go title="ent/schema/fridge.go"
+// Edges of the Fridge.
+func (Fridge) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("compartments", Compartment.Type).
+		    // Do not generate an endpoint for POST /fridges/{id}/compartments
+		    Annotation(entoas.CreateOperation(entoas.OperationPolicy(entoas.PolicyExclude))),
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:212]: line highlight

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+
+<div style={{textAlign: 'center'}}>
+  <img alt="One Schema per Endpoint" src="https://entgo.io/images/assets/elkopa/1.png" />
+  <p style={{fontSize: 12}}>One Schema per Endpoint</p>
+</div>
+
+Many users have requested to change this behaviour to simply map the Ent schema to the OAS document. Therefore, you now
+can configure `entoas` to do that: 
+
+```diff
```
Comment (rotemtam) [inline doc/website/blog/2021-11-10-announcing-entoas.md:243]: use `go` and line highlight the new line

**Review by rotemtam — CHANGES_REQUESTED** [review]
Great job @masseelch 
Very excited to have this released soon

**Comment by masseelch** [comment]
Note to me: Link to discord as well

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+	ex, err := entoas.NewExtension()
+	if err != nil {
+		log.Fatalf("creating entoas extension: %v", err)
+	}
+	err = entc.Generate("./schema", &gen.Config{}, entc.Extensions(ex))
+	if err != nil {
+		log.Fatalf("running ent codegen: %v", err)
+	}
+}
+
```
Comment (a8m) [inline doc/website/blog/2021-11-10-announcing-entoas.md:57]:
```suggestion
```

File: doc/website/blog/2021-11-10-announcing-entoas.md (hunk 9 above)
Comment (a8m) [reply doc/website/blog/2021-11-10-announcing-entoas.md:182]: Link to images from the previous post.

https://entgo.io/images/assets/elkopa/2.png

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+		    Annotation(entoas.CreateOperation(entoas.OperationPolicy(entoas.PolicyExclude))), 
+	}
+}
+
+// Annotations of the Fridge.
+func (Fridge) Annotations() []schema.Annotation {
+	return []schema.Annotation{
+	    // Do not generate an endpoint for DELETE /fridges/{id}
+		entoas.DeleteOperation(entoas.OperationPolicy(entoas.PolicyExclude)),
+	}
```
Comment (a8m) [inline doc/website/blog/2021-11-10-announcing-entoas.md:261]:
```suggestion
	return []schema.Annotation{
		// Do not generate an endpoint for DELETE /fridges/{id}
		entoas.DeleteOperation(entoas.OperationPolicy(entoas.PolicyExclude)),
	}
```

File: doc/website/blog/2021-11-10-announcing-entoas.md [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+Ent. This feature connects between Ent's code-generation capabilities and OpenAPI/Swagger's rich tooling ecosystem. 
+
+Have questions? Need help with getting started? Feel free to [join our Slack channel](https://entgo.io/docs/slack/).
+
+:::note For more Ent news and updates:
+
+- Subscribe to our [Newsletter](https://www.getrevue.co/profile/ent)
+- Follow us on [Twitter](https://twitter.com/entgo_io)
+- Join us on #ent on the [Gophers Slack](https://entgo.io/docs/slack)
+
```
Comment (a8m) [inline doc/website/blog/2021-11-10-announcing-entoas.md:309]: discord

**Review by a8m — APPROVED** [review]
Wonderful! Thanks for working on it Jannik!

**Comment by masseelch** [comment]
Thanks! @a8m Only images are left.

**Pull request merged**